addons/website_livechat/controllers/main.py

```diff
@@ -68,6 +72,14 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
             anonymous_name = _('Visitor #%s') % visitor_sudo.id
         return super(WebsiteLivechat, self).get_session(channel_id, anonymous_name, previous_operator_id=previous_operator_id, chatbot_script_id=chatbot_script_id, persisted=persisted, **kwargs)
 
+    @http.route()
+    def im_livechat_chat_history(self, uuid, last_id=False, limit=20):
+        result = super().im_livechat_chat_history(uuid, last_id=last_id, limit=limit)
+        channel = request.env["discuss.channel"].sudo().search([("uuid", "=", uuid)], limit=1)
+        if channel and request.env.user._is_public() and not channel.channel_member_ids.filtered(lambda member: member.guest_id):
+            self._add_current_persona_to_channel(channel.sudo(), post_joined_message=False)
```

Comment:
filtered guest should match current guest, otherwise something else is wrong

Reply:
This tried to do what `discuss_channel_add_guest_as_member` does I think.

Reply:
`discuss_channel_add_guest_as_member` is executed after the guest has been authenticated at least once (the guest information is available on the request) while this is not the case here. When the operator requests a chat with the current visitor, the guest is not yet added to the channel.